ImageGallery/ImageGallery/ImageStore 2.swift

```diff
@@ -0,0 +1,109 @@
+//
+//  DownloadStore.swift
+//  ImagaeGallery
+//
+//  Created by Nick Ryan on 6/18/24.
+//
+
+import SwiftUI
+import Combine
+import Foundation
+
+@MainActor
+class ImageStore: NSObject, ObservableObject, URLSessionDownloadDelegate {
+  @Published var photos: [PexelsPhoto] = []
+  
+  private var session: URLSession
+  private var progressHandler: ((Double) -> Void)?
+  private var downloadTask: URLSessionDownloadTask?
+  private var downloadCompletionHandler: ((URL) -> Void)?
+  
+  override init() {
+    let sessionConfiguration = URLSessionConfiguration.default
+    self.session = URLSession(configuration: sessionConfiguration, delegate: nil, delegateQueue: nil)
```

Comment:
Instead of creating a session here, you can just use `URLSession.shared` and set the delegate.